Automattic/jetpack
# Jetpack Connection: Add inline comments when whole input is processed

We are adding inline comments in the Connection package to explain cases where the whole `$_POST` input is used.
We are also refactoring `setup_xmlrpc_handlers` in `Manager` class to avoid using the whole `$_POST` input.

## Proposed changes:

* `Jetpack_Signature::sign_current_request`: We need the whole `$_POST` in order to verify a cryptographic signature of the post data.
* `Automattic\Jetpack\Connection\Manager::internal_verify_xml_rpc_signature`: We need the whole `$_POST` in order to verify a cryptographic signature of the post data.
* `Automattic\Jetpack\Connection\Manager::setup_xmlrpc_handlers`: Deprecate the `$request_params` method argument and check for `$_GET['for']` and `$_GET['jetpack']` inside the method instead. I've audited the usage of this method and confirmed it's used only by the same class.

### Other information:


## Jetpack product discussion


p9dueE-8ca-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






- For the inline comments: Code Review: Do you believe we need further explanation?
- For the change in `setup_xmlrpc_handlers`: I believe the best way to test this change (apart from the code review) would be to use the JP Debugger and confirm the same results for a given site with/without the PR applied. The JP Debugger will make a bunch of REST and XML-RPC requests to the site with/without authentication too. Also make sure to set the site to use the alternate XML-RPC url in order to set the `comms` param is properly checked too.
